torchvision/models/regnet.py

```diff
@@ -0,0 +1,785 @@
+        stage_index: int = 0,
+    ) -> None:
+        super().__init__()
+        self.stage_depth = 0
+
+        for i in range(depth):
+            block = block_constructor(
+                width_in if i == 0 else width_out,
+                width_out,
+                stride if i == 0 else 1,
+                bn_epsilon,
+                bn_momentum,
+                activation,
+                group_width,
+                bottleneck_multiplier,
+                se_ratio,
+            )
+
+            self.stage_depth += block.depth
+            self.add_module(f"block{stage_index}-{i}", block)
+
+
+class RegNetParams:
+    def __init__(
+        self,
+        depth: int,
+        w_0: int,
+        w_a: float,
+        w_m: float,
+        group_width: int,
+        bottleneck_multiplier: float = 1.0,
+        stem_type: Callable[..., nn.Module] = SimpleStemIN,
+        stem_width: int = 32,
+        block_type: Callable[..., nn.Module] = ResBottleneckBlock,
+        activation: Callable[..., nn.Module] = nn.ReLU,
+        use_se: bool = True,
+        se_ratio: float = 0.25,
+        bn_epsilon: float = 1e-05,
+        bn_momentum: float = 0.1,
+        num_classes: int = 1000,
+    ) -> None:
+        if w_a < 0 or w_0 <= 0 or w_m <= 1 or w_0 % 8 != 0:
+            raise ValueError("Invalid RegNet settings")
+        self.depth = depth
+        self.w_0 = w_0
+        self.w_a = w_a
+        self.w_m = w_m
+        self.group_width = group_width
+        self.bottleneck_multiplier = bottleneck_multiplier
+        self.stem_type = stem_type
+        self.block_type = block_type
+        self.activation = activation
+        self.stem_width = stem_width
+        self.use_se = use_se
+        self.se_ratio = se_ratio if use_se else None
+        self.bn_epsilon = bn_epsilon
+        self.bn_momentum = bn_momentum
+        self.num_classes = num_classes
+
+    def get_expanded_params(self):
+        """
+        Programatically compute all the per-block settings,
+        given the RegNet parameters.
+
+        The first step is to compute the quantized linear block parameters,
+        in log space. Key parameters are:
+        - `w_a` is the width progression slope
+        - `w_0` is the initial width
+        - `w_m` is the width stepping in the log space
+
+        In other terms
+        `log(block_width) = log(w_0) + w_m * block_capacity`,
+        with `bock_capacity` ramping up following the w_0 and w_a params.
+        This block width is finally quantized to multiples of 8.
+
+        The second step is to compute the parameters per stage,
+        taking into account the skip connection and the final 1x1 convolutions.
+        We use the fact that the output width is constant within a stage.
+        """
+
+        QUANT = 8
+        STRIDE = 2
+
+        # Compute the block widths. Each stage has one unique block width
+        widths_cont = np.arange(self.depth) * self.w_a + self.w_0
```

Comment:
not super important, but would it be possible to use `torch` functions instead of `np` functioned whenever possible?